### PR TITLE
MSVC C++17

### PR DESCRIPTION
**Pull request by dvirtz**
Added toolchains for VS 15 2017 with C++17 switch

### CODE REVIEW

File: flags/cxx17.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+endif()
+
+include(polly_add_cache_flag)
+
+string(COMPARE EQUAL "${ANDROID_NDK_VERSION}" "" _not_android)
+
+
+# TODO: test other platfroms, CMAKE_CXX_FLAGS_INIT should work for all
+if(HUNTER_CMAKE_GENERATOR MATCHES "^Visual Studio.*$")
+  polly_add_cache_flag(CMAKE_CXX_FLAGS "/std:c++17")
```
Comment (ruslo) [inline flags/cxx17.cmake]: I think this trick is not working with Visual Studio generator. That's why all flags listed explicitly in `vs-mt` for example:
* https://github.com/ruslo/polly/blob/b9819d2e7f2be813b97654f6cbba20831751fa58/flags/vs-mt.cmake#L11-L14

File: vs-15-2017-cxx17.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2015-2017, Ruslan Baratov
```
Comment (ruslo) [inline vs-15-2017-cxx17.cmake]: Please add it to toolchain table:
* https://github.com/ruslo/polly/blob/b9819d2e7f2be813b97654f6cbba20831751fa58/bin/detail/toolchain_table.py#L180-L185

**Comment by dvirtz** [comment]
Using `CMAKE_CXX_FLAGS_INIT` instead of `CMAKE_CXX_FLAGS` seems to fix it.

**Comment by dvirtz** [comment]
Thanks

**Pull request merged**